crates/types/src/vote.rs

```diff
@@ -721,249 +820,11 @@ impl<TYPES: NodeType, COMMITMENT: CommitmentBounds, VOTE: VoteType<TYPES, COMMIT
-    Accumulator2<TYPES, COMMITMENT, VOTE> for AccumulatorPlaceholder<TYPES, COMMITMENT, VOTE>
-{
-    fn append(
-        self,
-        _vote: VOTE,
-        _vote_node_id: usize,
-        _stake_table_entries: Vec<<TYPES::SignatureKey as SignatureKey>::StakeTableEntry>,
-    ) -> Either<Self, AssembledSignature<TYPES>> {
-        either::Left(self)
-    }
-}
-
 /// Mapping of commitments to vote tokens by key.
-// TODO ED Remove this whole token generic
 type VoteMap<COMMITMENT, TOKEN> = HashMap<
     COMMITMENT,
     (
         u64,
         BTreeMap<EncodedPublicKey, (EncodedSignature, VoteData<COMMITMENT>, TOKEN)>,
     ),
 >;
-
-/// Describe the process of collecting signatures on block or leaf commitment, to form a DAC or QC,
-/// respectively.
-///
-/// TODO GG used only in election.rs; move this to there and make it private?
-pub struct VoteAccumulator<TOKEN, COMMITMENT: CommitmentBounds, TYPES: NodeType> {
-    /// Map of all signatures accumlated so far
-    pub total_vote_outcomes: VoteMap<COMMITMENT, TOKEN>,
-    /// Map of all da signatures accumlated so far
-    pub da_vote_outcomes: VoteMap<COMMITMENT, TOKEN>,
-    /// Map of all yes signatures accumlated so far
-    pub yes_vote_outcomes: VoteMap<COMMITMENT, TOKEN>,
-    /// Map of all no signatures accumlated so far
-    pub no_vote_outcomes: VoteMap<COMMITMENT, TOKEN>,
-    /// Map of all view sync precommit votes accumulated thus far
-    pub viewsync_precommit_vote_outcomes: VoteMap<COMMITMENT, TOKEN>,
-    /// Map of all view sync commit votes accumulated thus far
-    pub viewsync_commit_vote_outcomes: VoteMap<COMMITMENT, TOKEN>,
-    /// Map of all view sync finalize votes accumulated thus far
-    pub viewsync_finalize_vote_outcomes: VoteMap<COMMITMENT, TOKEN>,
-    /// A quorum's worth of stake, generall 2f + 1
-    pub success_threshold: NonZeroU64,
-    /// Enough stake to know that we cannot possibly get a quorum, generally f + 1
-    pub failure_threshold: NonZeroU64,
-    /// A list of valid signatures for certificate aggregation
-    pub sig_lists: Vec<<TYPES::SignatureKey as SignatureKey>::PureAssembledSignatureType>,
-    /// A bitvec to indicate which node is active and send out a valid signature for certificate aggregation, this automatically do uniqueness check
-    pub signers: BitVec,
-}
-
-impl<TOKEN, COMMITMENT: CommitmentBounds, TYPES: NodeType>
-    Accumulator<
-        (
-            COMMITMENT,
-            (
-                EncodedPublicKey,
-                (
-                    EncodedSignature,
-                    Vec<<TYPES::SignatureKey as SignatureKey>::StakeTableEntry>,
-                    usize,
-                    VoteData<COMMITMENT>,
-                    TOKEN,
-                ),
-            ),
-        ),
-        AssembledSignature<TYPES>,
-    > for VoteAccumulator<TOKEN, COMMITMENT, TYPES>
-where
-    TOKEN: Clone + VoteToken,
-{
-    #![allow(clippy::too_many_lines)]
-    fn append(
```

Comment:
I see append implemented for TimeCertificate, did we already have it for the other certificate types, I can't find it in this PR and am confused since it obviously works

Reply:
Yes, they were in a previous PR!